File: paymentMicroservice/src/main/java/com/paymentMicroservice/dto/CreatePaymentItemDTO.java
package com.paymentMicroservice.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;

import java.util.UUID;

@Getter
@Schema(name = "CreatePaymentItemDTO", description = "DTO representing a single item in a payment")
public class CreatePaymentItemDTO {

    @Schema(description = "Unique identifier for the payment item", example = "item_12345", nullable = true)
    private String id;

    @Schema(description = "Amount associated with the payment item. Must be at least 1.",
            example = "5000", required = true)
    @NotNull
    @Min(1)
    private Long amount;

    @Schema(description = "Budget Subtype ID associated with the item (if applicable)",
            example = "d290f1ee-6c54-4b01-90e6-d701748f0851", nullable = true)
    private UUID budgetSubtypeId;

    @Schema(description = "Budget Type ID associated with the item (if applicable)",
            example = "c123d456-78e9-0123-4567-890ab123cdef", nullable = true)
    private UUID budgetTypeId;

    @Schema(description = "Supplier ID associated with the item",
            example = "f123a456-78e9-0123-4567-890cd123efab", required = true)
    @NotNull
    private UUID supplierId;

    @Schema(description = "IVA rate applied to the item (default is 0.0)",
            example = "0.23", nullable = true)
    private Double ivaRate = 0.0;
}

File: client/src/app/services/models/create-payment-item-dto.ts
/**
 * DTO representing a single item in a payment
 */
export interface CreatePaymentItemDto {

  /**
   * Amount associated with the payment item. Must be at least 1.
   */
  amount: number;

  /**
   * Budget Subtype ID associated with the item (if applicable)
   */
  budgetSubtypeId?: string | null;

  /**
   * Budget Type ID associated with the item (if applicable)
   */
  budgetTypeId?: string | null;

  /**
   * Unique identifier for the payment item
   */
  id?: string | null;

  /**
   * IVA rate applied to the item (default is 0.0)
   */
  ivaRate?: number | null;

  /**
   * Supplier ID associated with the item
   */
  supplierId: string;
}
